hypothesis-python/src/hypothesis/strategies/_internal/datetime.py

```diff
@@ -29,6 +39,28 @@ def is_pytz_timezone(tz):
     return module == "pytz" or module.startswith("pytz.")
 
 
+def draw_capped_multipart(data, min_value, max_value):
+    assert isinstance(min_value, (dt.date, dt.time, dt.datetime))
+    assert type(min_value) == type(max_value)
+    assert min_value <= max_value
+    result = {}
+    cap_low, cap_high = True, True
+    names = {dt.date: DATENAMES, dt.time: TIMENAMES, dt.datetime: DATENAMES + TIMENAMES}
+    for name in names[type(min_value)]:
```

Comment:
nit: could we come up with a better pair of variable names here? I always get twitchy when I see a pair of variable names that are a single character apart.